newrelic-agent/src/main/java/com/newrelic/api/agent/NewRelicApiImplementation.java

```diff
@@ -56,7 +58,9 @@ public NewRelicApiImplementation() {
      */
     @Override
     public void noticeError(Throwable throwable, Map<String, ?> params) {
-        noticeError(throwable, params, false);
+        noticeError(throwable, params, throwable != null &&
+                        isExpectedErrorConfigured(throwable.getClass().getName(), throwable.getMessage()),
```

Comment:
Maybe now I can just pass the throwable here and do the messy logic inside `isExpectedErrorConfigured()`
Would that make more sense or is this okay?